nativelib/src/main/resources/gc/commix/datastructures/GreyPacket.c

```diff
@@ -0,0 +1,115 @@
+#include "../Object.h"
+#include "GreyPacket.h"
+#include "../Log.h"
+#include <string.h>
+
+bool GreyPacket_Push(GreyPacket *packet, Stack_Type value) {
+    assert(value != NULL);
+    if (packet->size >= GREY_PACKET_ITEMS) {
+        return false;
+    } else {
+        packet->items[packet->size++] = value;
+        return true;
+    }
+}
+Stack_Type GreyPacket_Pop(GreyPacket *packet) {
+    assert(packet->size > 0);
+    return packet->items[--packet->size];
+}
+
+void GreyPacket_Move(GreyPacket *src, GreyPacket *dst, int count) {
```

Comment:
It looks like GreyPacket_Split would be a better name for this.

Reply:
How about `GreyPacket_MoveItems` ? I realize `GreyPacket_Move` does not work because it does not move the entire packet. `GreyPacket_Split` also does not fit well because we do not take a packet and return two or more packets.

Reply:
`MoveItems` is much better than just `Move`.